analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,924 @@
+
+.. data:: orderTax
+
+    **Optional** Order tax amount written as number - not string.
+
+.. data:: orderShipping
+
+    **Optional** Order shipping costs, written as number - not string.
+
+.. data:: orderDiscount
+
+    **Optional** Order discount amount, written as number - not string.
+
+Example of usage::
+
+    _paq.push(["trackEcommerceOrder", "3352", 499, 399, 0, 100]);
+
+Updating cart
+`````````````
+.. todo::
+
+Why Tracker doesn't count cartAmount by itself? Why user must do this?
+
+To update user cart (when user adds new product or removes them from cart) use ``trackEcommerceCartUpdate`` function::
+
+    _paq.push(["trackEcommerceCartUpdate", cartAmount]);
+
+.. data:: cartAmount
+
+    **Required** Cart amount (sum of products), written as number.
+
+
+.. warning::
+
+    Before tracking cart update be sure you added all products in cart by using ``addEcommerceItem`` first.
+    (Even ones that were in cart earlier) Then use this function last.
+
+Example of usage::
+
+        _paq.push(["trackEcommerceCartUpdate", 250]);
+
+Tracking product / category view
+````````````````````````````````
+If you want to track when user enters product site, or is browsing products category use ``setEcommerceView`` function::
+
+    _paq.push(["setEcommerceView", productSKU, productName, productCategory, productPrice]);
+
+.. data:: productSKU
+
+    **Required** String with product stock-keeping unit. False for tracking category.
+
+.. data:: productName
+
+    **Optional** String with product name. False for tracking category.
+
+.. data:: productCategory
+
+    **Optional** Product category, can be written as Array with up to 5 elements.
+
+.. data:: productPrice
+
+    **Optional** Float with product price.
+
+.. warning::
+
+    All string parameters should be escaped, without double quote character.
+
+Example of usage::
+
+    _paq.push(["setEcommerceView", "craft-311", "Unicorn Iron on Patch", "Crafts & Sewing", "499"]);
+
+
+Custom Variables
+^^^^^^^^^^^^^^^^
+.. todo::
+What's difference between custom variables and dimensions? Maybe some sort of help.center link?
+
+
+Adding / Editing Custom Variable
+````````````````````````````````
+.. todo::
+
+Is this variable set in the portal / using API first, then I can use the id slot to define it's name and value?+
+What's name used for? Can it be accessed later?
+
+
+To set custom variable that can be used later, use ``setCustomVariable`` function::
+
+    _paq.push(["setCustomVariable", index, name, value, scope]);
+
+.. data:: index
+
+    **Required** Number from 1 to 5 where variable is stored
+
+.. data:: name
+
+   **Required** Name of the variable
+
+.. data:: value
+
+   **Required** Value of the variable limited to 200 characters.
+
+.. data:: scope
+
+   **Required** Scope of the variable, "visit" or "page"
+
+
+.. warning::
+
+    Custom Variable with scope set on "visit" will be saved for visit, you don't need to save it every page.
```

Comment:
I think this should be a `note` to the `scope` description. I think you can indent it to make it so.
It would be better to specify how both of those values affect availability of the variable (visit - remembered until end of visit/session [we should make a term for that in glossary to avoid confusion], page - remembered until page is changed).

It's also not clear to me if visit and session are the same in this case. Piwik lets admin set visit max length, but I'm not sure if visit variable is deleted when visit Piwik ends. It's probably stored in session cookie. Can we request clarification in TODO?